### ISSUE TITLE
(aws-events-targets): (Add support for Predefined variables for Input transformers)

### ISSUE THREAD

**Issue by luisMolina95** [aws/aws-cdk#27841]
### Describe the feature

Currently there is no available feature to make use of  [Predefined variables](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-transform-target-input.html#eb-transform-input-predefined).

**AWSEventField** would work the same way as **EventField** but for the **AWS Predefined variables** ( `aws.events.rule-arn`, `aws.events.event.json`, `aws.events.rule-name`, `aws.events.event.ingestion-time`, `aws.events.event` ).

### Use Case

Without **Predefined variables** available in CDK, we need to keep deploying changes to the input transformer, and keep manually adding attributes with RuleTargetInput.

When we could instead just use `aws.events.event.json` to add the original event and then add the additional metadata but rigth now there is no option to use **Predefined variables** at all.

### Proposed Solution

For example this code:
```typescript
import { EventField, AWSEventField } from "aws-cdk-lib/aws-events";

const ruleTargetInput = RuleTargetInput.fromObject({
      // Currently this does not work, it is set up as a constant and not an Input Transformer
      // ruleName: "<aws.events.rule-name>",
      // originalEvent: "<aws.events.event.json>",
      // This new feature would make it possible:
      ruleName: AWSEventField.ruleName,
      originalEvent: AWSEventField.eventJSON ,
    });
```
Would generate the following stack template (an empty object input map and an input template with only Predefined variables):
```json
{
  "Arn": {
    "Fn::GetAtt": ["function", "Arn"]
  },
  "Id": "Target0",
  "InputTransformer": {
    "InputPathsMap": { },
    "InputTemplate": "{\"ruleName\":<aws.events.rule-name>,\"originalEvent\":<aws.events.event.json>}"
  }
}
```

### Other Information

https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-transform-target-input.html#eb-transform-input-predefined

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.104.0

### Environment details (OS name and version, etc.)

Windows 10

**Comment by RMSD** [comment]
Was there every any change on this request? It seems nice to have.

**Comment by luisMolina95** [comment]
> Was there every any change on this request? It seems nice to have.

I think they forgot about it 😅.

It's not the same but way back I found a workaround to do something similar.

https://github.com/aws/aws-cdk/discussions/27833

**Comment by RMSD** [comment]
Yep, I actually found that discussion after I continued to search and am using the workaround. Works great for now, even though it looks odd. Thought I'd bring some eyes to this though.
